### PR TITLE
Compute `BUILD_IMAGE`, `DOCKER_FILE`, `UCX_PY_VER` in build script

### LINKED ISSUE #6: Compute `BUILD_IMAGE` and `UCX_PY_VER` in build script
Because we have separate matrix columns for `BUILD_IMAGE` and `UCX_PY_VER`, whenever the matrix is updated with a new value for `DOCKER_FILE` or `RAPIDS_VER` we must update the `excludes` accordingly to ensure that we aren't running any invalid builds (i.e. publishing `dask.Dockerfile` to `gpuci/distributed` or using `cudf=21.10` with `ucx-py=0.21`).

As seen with #5, this means we must manually write out several excludes rules that have a large impact on CI across several repos (as getting these wrong essentially means all the gpuCI images will be broken). I think a simpler way of handling this would be to remove `BUILD_IMAGE` and `UCX_PY_VER` from the matrix altogether, replacing them with values computed within the build script:

- For `DOCKER_FILE=<file_name>.Dockerfile`, we can set `BUILD_IMAGE=gpuci/<file_name>`
- Mapping / case statements can be used / added manually to set `UCX_PY_VER` using the value of `RAPIDS_VER`; though this still must be updated manually, it would be a simpler change than those made to the exclusions

### PR DESCRIPTION
**Pull request by charlesbluca**
Closes #6 

This allows us to remove the `excludes` in the matrix file, which should simplify the process of adding a Dockerfile to this repo

### CODE REVIEW

File: ci/axis/dask.yaml [hunk 1]
```diff
@@ -21,27 +16,4 @@ RAPIDS_VER:
...
-    BUILD_IMAGE: gpuci/dask
-  - DOCKER_FILE: dask_sql.Dockerfile
-    BUILD_IMAGE: gpuci/distributed
-
-  - RAPIDS_VER: '21.08'
-    UCX_PY_VER: '0.22'
-  - RAPIDS_VER: '21.10'
-    UCX_PY_VER: '0.21'
\ No newline at end of file
+excludes:
```
Comment (pentschev) [inline ci/axis/dask.yaml:19]: Is it necessary to have an empty `excludes` section or could it be removed?

**Review by pentschev — COMMENTED** [review]
Looks good to me, tagging @raydouglass here who may have comments too.

File: ci/axis/dask.yaml (hunk 1 above)
Comment (charlesbluca) [reply ci/axis/dask.yaml:19]: Not sure - my assumption was that it should be included, as we also include it in the [axis file](https://github.com/dask/dask/blob/main/continuous_integration/gpuci/axis.yaml) for Dask's gpuCI, but ops should be able to answer this question.

Comment (pentschev) [reply ci/axis/dask.yaml:19]: @raydouglass do you know if an empty `excludes` section is mandatory or if we could remove it?

File: ci/gpuci/run.sh [hunk 2]
```diff
@@ -21,9 +21,23 @@ echo $DH_TOKEN | docker login --username $DH_USER --password-stdin &> /dev/null
 BUILD_TAG="${RAPIDS_VER}-cuda${CUDA_VER}-devel-${LINUX_VER}-py${PYTHON_VER}"
 
 # Setup BUILD_ARGS
+case $RAPIDS_VER in
```
Comment (quasiben) [inline ci/gpuci/run.sh:24]: When we need to update, is this the only section which requires modification ?

Comment (charlesbluca) [reply ci/gpuci/run.sh:24]: This and the `RAPIDS_VER` block in the axis file:

https://github.com/rapidsai/dask-build-environment/blob/79e1b8bd794d37f24aede7ed1aa28b32ec3d4a47/ci/axis/dask.yaml#L20-L22

**Review by quasiben — APPROVED** [review]
LGTM.  We should also here from @raydouglass before merging in

File: ci/axis/dask.yaml (hunk 1 above)
Comment (raydouglass) [reply ci/axis/dask.yaml]: It is mandatory, it just be left empty though

**Review by raydouglass — APPROVED** [review]

**Comment by raydouglass** [comment]
LGTM

Just throwing this out there: you can use [yaml-axis-parser](https://pypi.org/project/yaml-axis-parser/) to compute the combinations that will be produced.

**Comment by charlesbluca** [comment]
Thanks @raydouglass! That should be helpful moving forward

**Pull request merged**